src/yaml-tools.js

```diff
@@ -111,11 +123,30 @@ yamlLibrary = {
   // doc: The SAM.YML parsed document
   // schema: the JSON schema to compare the doc to
   checkDocAgainstSchema: function (doc, schema, failed, warning, info) {
```

Comment:
This first argument might be better as something like docIndex instead of doc.

Reply:
I was on the fence about that, too.  I'll change it for clarity.